src/core/ext/filters/http/server/http_server_filter.cc

```diff
@@ -301,7 +304,15 @@ static void hs_recv_initial_metadata_ready(void* user_data, grpc_error* err) {
   } else {
     GRPC_ERROR_REF(err);
   }
-  GRPC_CLOSURE_RUN(calld->original_recv_initial_metadata_ready, err);
+  grpc_closure* closure = calld->original_recv_initial_metadata_ready;
```

Comment:
No need to do the two-phase thing for unsetting `original_recv_initial_metadata_ready` here, since we can use `seen_recv_initial_metadata_ready` below.